tracing-core/src/dispatch.rs

```diff
@@ -859,6 +884,45 @@ where
     }
 }
 
+impl WeakDispatch {
+    /// Attempts to upgrade the `WeakDispatch` to a `Dispatch`.
```

Comment:
it might be nice if this explained the situations under which this can and cannot be upgraded --- we can probably copy from the `std::sync::Weak` docs.

Reply:
Just pushed: https://github.com/tokio-rs/tracing/pull/2293/commits/d50d6f862bf9c7c1d257cc542777cbda2921974e